app/jobs/regular/create_linked_topic.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+module Jobs
+  class CreateLinkedTopic < ::Jobs::Base
+
+    def execute(args)
+      reference_post = Post.find_by(id: args[:post_id])
+      return unless reference_post.present?
+      parent_topic = reference_post.topic
+      return unless parent_topic.present?
+      parent_topic_id = parent_topic.id
+      parent_title = parent_topic.title
+      title_suffix_locale = I18n.t('create_linked_topic.title_suffix')
+
+      linked_topic_record = parent_topic.linked_topic
+      if linked_topic_record.present?
+        raw_title = parent_title.gsub(/\(#{title_suffix_locale} (\d+)\)$/i, "").strip
+        original_topic_id = linked_topic_record.original_topic_id
+        sequence = linked_topic_record.sequence + 1
+      else
+        raw_title = parent_title
+
+        # update parent topic title to append title_suffix_locale
+        parent_title = "#{parent_title} (#{title_suffix_locale} 1)"
```

Comment:
We should use a transaction for this stuff. We wouldn't want a title updated if something fails below.

Probably most of this job should be in a big transaction.

Reply:
Noted, will do.